hssi 10G: port_ -> src_port_ and dst_port_

tswhison opened the pull request:
Replaces the notion of a single port for both Tx/Rx with one port for the Tx side and a separate port for the Rx side.

Adds mbox_write(), mbox_read() variants that accept a port selection as a parameter.

Commit:
hssi 10G: port_ -> src_port_ and dst_port_

Replaces the notion of a single port for both Tx/Rx with
one port for the Tx side and a separate port for the Rx
side.

Adds mbox_write(), mbox_read() variants that accept a
port selection as a parameter.

Signed-off-by: Tim Whisonant <tim.whisonant@intel.com>

anandaravuri reviewed (dismissed):
Can you add Tx and Rx Port changes to 100G also ?

tswhison commented:
Yes, but I'd like to save that for a separate pull request.

Commit:
hssi 10G: add back port_

Initialize port_ to an invalid port number at the start of time.
In the run function, check to see whether a valid value was given for
port_. If so, then update src_port_ and dst_port_ to that value.

Signed-off-by: Tim Whisonant <tim.whisonant@intel.com>

anandaravuri commented on samples/hssi/hssi_10g_cmd.h:
```diff
@@ -159,8 +170,13 @@ class hssi_10g_cmd : public hssi_cmd
     if (eth_ifc_ == "none")
         eth_ifc = hafu->ethernet_interface();
 
+    if (port_ != INVALID_PORT)
```
corner case:  if user pass all port_, dst_port_,  src_port_ values  HSSI tool overrides  dst_port_,  src_port_ values.

tswhison replied:
Yes, would you like me to also print a message in this case?

anandaravuri replied:
Print message and continue

Commit:
Address review feedback

Signed-off-by: Tim Whisonant <tim.whisonant@intel.com>

aditya-patro commented on samples/hssi/hssi_10g_cmd.h:
```diff
@@ -352,20 +356,20 @@ class hssi_10g_cmd : public hssi_cmd
+      int_to_hex(hafu->mbox_read(src_port_, CSR_TX_END_TSTAMP)) << std::endl;
   
     os << "0x3d00 " << std::setw(22) << "num_pkt" << ": " <<
-      int_to_hex(hafu->mbox_read(CSR_NUM_PKT)) << std::endl;
```
0x3d00 to 0x3dff all csr access are for dst_port

tswhison replied:
Got it, thanks! 👍

Commit:
Address review comments.

Signed-off-by: Tim Whisonant <tim.whisonant@intel.com>

Commit:
Merge branch 'master' into tswhison/hssi_update

Commit:
hssi: apply patch for CSR_TX_STA_TSTAMP

Signed-off-by: Tim Whisonant <tim.whisonant@intel.com>

Commit:
Merge branch 'master' into tswhison/hssi_update
